pkg/kt/snmp.go

```diff
@@ -218,20 +220,26 @@ type SnmpDiscoConfig struct {
 	IgnoreOrig         string          `yaml:"-"`
 }
 
+type ProviderMap struct {
+	UserTags  map[string]string `yaml:"user_tags"`
+	MatchAttr map[string]string `yaml:"match_attributes"`
+}
+
 type SnmpGlobalConfig struct {
-	PollTimeSec   int               `yaml:"poll_time_sec"`
-	DropIfOutside bool              `yaml:"drop_if_outside_poll"`
-	MibProfileDir string            `yaml:"mib_profile_dir"`
-	MibDB         string            `yaml:"mibs_db"`
-	MibsEnabled   []string          `yaml:"mibs_enabled"`
-	TimeoutMS     int               `yaml:"timeout_ms"`
-	Retries       int               `yaml:"retries"`
-	GlobalV3      *V3SNMPConfig     `yaml:"global_v3"`
-	RunPing       bool              `yaml:"response_time"`
-	PingSec       int               `yaml:"ping_interval_sec,omitempty"`
-	PurgeDevices  int               `yaml:"purge_devices_after_num"` // Delete any device if its not seen after X discovery attempts. Default is 0, which means things never get purged.
-	UserTags      map[string]string `yaml:"user_tags"`
-	MatchAttr     map[string]string `yaml:"match_attributes"`
+	PollTimeSec   int                    `yaml:"poll_time_sec"`
+	DropIfOutside bool                   `yaml:"drop_if_outside_poll"`
+	MibProfileDir string                 `yaml:"mib_profile_dir"`
+	MibDB         string                 `yaml:"mibs_db"`
+	MibsEnabled   []string               `yaml:"mibs_enabled"`
+	TimeoutMS     int                    `yaml:"timeout_ms"`
+	Retries       int                    `yaml:"retries"`
+	GlobalV3      *V3SNMPConfig          `yaml:"global_v3"`
+	RunPing       bool                   `yaml:"response_time"`
+	PingSec       int                    `yaml:"ping_interval_sec,omitempty"`
+	PurgeDevices  int                    `yaml:"purge_devices_after_num"` // Delete any device if its not seen after X discovery attempts. Default is 0, which means things never get purged.
+	UserTags      map[string]string      `yaml:"user_tags"`
+	MatchAttr     map[string]string      `yaml:"match_attributes"`
+	ProviderMap   map[string]ProviderMap `yaml:"provider_maps"`
```

Comment:
Nit: Not a big deal, but maybe should name this `providers` instead of `provider_maps` to be consistent with `user_tags` / `match_attributes`